runtime/command-metrics/src/main/java/io/aklivity/zilla/runtime/command/metrics/internal/airline/ZillaMetricsCommand.java

```diff
@@ -128,67 +78,68 @@ private Path engineDirectory()
                 rethrowUnchecked(ex);
             }
         }
-        EngineConfiguration config = new EngineConfiguration(props);
-        return config.directory();
-    }
-
-    private List<MetricsLayout> countersLayouts() throws IOException
-    {
-        Stream<Path> files = Files.walk(metricsPath, 1);
-        return files.filter(this::isCountersFile).map(this::newCountersLayout).collect(toList());
-    }
-
-    private List<MetricsLayout> gaugesLayouts() throws IOException
-    {
-        Stream<Path> files = Files.walk(metricsPath, 1);
-        return files.filter(this::isGaugesFile).map(this::newGaugesLayout).collect(toList());
-    }
-
-    private List<MetricsLayout> histogramsLayouts() throws IOException
-    {
-        Stream<Path> files = Files.walk(metricsPath, 1);
-        return files.filter(this::isHistogramsFile).map(this::newHistogramsLayout).collect(toList());
-    }
+        final Configuration config = new EngineConfiguration(props);
+        final ErrorHandler onError = Throwable::printStackTrace;
+        final String binding = args != null && args.size() >= 1 ? args.get(0) : null;
 
-    private boolean isCountersFile(
-        Path path)
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                COUNTERS_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
-    }
-
-    private boolean isGaugesFile(
-        Path path)
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                GAUGES_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
-    }
-
-    private boolean isHistogramsFile(
-        Path path)
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                HISTOGRAMS_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
+        do
+        {
+            try (Engine engine = Engine.builder()
+                .config(config)
+                .errorHandler(onError)
+                .readonly()
+                .build())
+            {
+                engine.start();
+                MetricsReader metrics = new MetricsReader(engine, engine::supplyLocalName);
+                LongPredicate filterPredicate = filterPredicate(namespace, binding, engine::supplyLabelId);
```

Comment:
Please rename `filterPredicate` variable to `filter`.

Reply:
done